mydemoproject/demoapp1/views.py

```diff
@@ -1,17 +1,186 @@
-from django.shortcuts import render
-from django.http import HttpResponse
+from django.contrib import messages
+from django.shortcuts import redirect
+from django.views.generic import (ListView, DeleteView,
+                                  UpdateView, CreateView,
+                                  TemplateView)
+from django.urls import reverse_lazy
+from .models.book import Book
+from .models.inventory import Inventory
+from .models.user import CustomUser
+from .models.borrowed_book import BorrowedBook
 
-# Create your views here.
 
-def show_books(request):
-    return render(request, 'demoapp1/books.html')
+# ListView for Users
+class UserListView(ListView):
+    model = CustomUser
+    template_name = 'demoapp1/users.html'
+    context_object_name = 'users'
 
-def show_users(request):
-    return render(request, 'demoapp1/users.html')
 
+# DeleteView for Users
+class UserDeleteView(DeleteView):
+    model = CustomUser
+    success_url = reverse_lazy('show_user')
+    pk_url_kwarg = 'user_id'
 
-def show_inventory(request):
-    return render(request, 'demoapp1/inventory.html')
 
-def show_borrowed(request):     
-    return render(request, 'demoapp1/borrowed.html')
+# UpdateView for User
+class UserUpdateView(UpdateView):
+    model = CustomUser
+    fields = ['username', 'email']
+    template_name = 'demoapp1/edit_form.html'
+    context_object_name = 'user'
+    success_url = reverse_lazy('show_user')
+    pk_url_kwarg = 'user_id'
+
+    def get_context_data(self, **kwargs):
```

Comment:
Why override this method? Please add comments.

Reply:
to add custom context, i am adding custom context i.e
`        context['edit_url'] = 'edit_book'` here